src/ocamlorg_static/file.ml

```diff
@@ -0,0 +1,18 @@
+let to_url_path ?digest filepath =
+  match digest with
+  | None -> filepath
+  | Some digest -> Fmt.str "/_/%s/%s" digest filepath
+
+type t = { filepath : string; digest : string option }
+
+let of_url_path path =
+  let xs = String.split_on_char '/' path in
+  match xs with
+  | [] -> raise (Invalid_argument "invalid Digest_url: path cannot be empty")
+  | "_" :: [] | [ "_"; _ ] ->
+      raise
+        (Invalid_argument
+           "invalid Digest_url: paths starting with '_' must be followed by a \
+            digest and filepath")
+  | "_" :: x :: xs -> { digest = Some x; filepath = String.concat "/" xs }
```

Comment:
Are you sure we can't simplify this one? Usually, sorting match branches in decreasing pattern length order allow maximum sharing. Something like that:
 
```suggestion
  | "_" :: x :: xs -> { digest = Some x; filepath = String.concat "/" xs }
  | "_" :: _ ->
      raise
        (Invalid_argument
           "invalid Digest_url: paths starting with '_' must be followed by a \
            digest and filepath")
```

Reply:
Thank you, good suggestion!